src/components/utils/rqb-inputs/custom-react-query-builder.tsx

```diff
@@ -0,0 +1,95 @@
+import { ErrorInput, FieldErrorAlert } from '@gridsuite/commons-ui';
+import { useIntl } from 'react-intl';
+import { combinatorType } from '../../dialogs/filter/expert/expert-filter-constants';
+import { useFormContext } from 'react-hook-form';
+import RemoveButton from 'components/utils/rqb-inputs/remove-button';
+import CombinatorSelector from 'components/utils/rqb-inputs/combinator-selector';
+import AddButton from 'components/utils/rqb-inputs/add-button';
+import ValueEditor from './value-editor';
+import { EXPERT_FILTER_QUERY } from '../../dialogs/filter/expert/expert-filter-form';
+
+interface ExpertFilterRQBProps {
+    name: string;
+    fields: Field[];
+}
+
+const CustomReactQueryBuilder = (props: ExpertFilterRQBProps) => {
+    const { getValues, setValue, watch } = useFormContext();
+    const intl = useIntl();
+
+    const query = watch(props.name);
+
+    const combinators: OptionList = [combinatorType.AND, combinatorType.OR].map(
+        (c) => ({
+            name: c.name,
+            label: intl.formatMessage({ id: c.label }),
+        })
+    );
+
+    const handleQueryChange = (newQuery: RuleGroupTypeAny) => {
+        const hasChanged =
+            formatQuery(getValues(EXPERT_FILTER_QUERY), 'json_without_ids') !==
+            formatQuery(newQuery, 'json_without_ids');
+        setValue(props.name, newQuery, { shouldDirty: hasChanged });
+    };
+
+    return (
+        <>
+            <Grid item xs={12}>
+                <QueryBuilderMaterial>
+                    <QueryBuilder
+                        fields={props.fields}
+                        query={query}
+                        addRuleToNewGroups={true}
+                        onQueryChange={handleQueryChange}
+                        getOperators={(fieldName) =>
+                            getOperators(fieldName, intl)
+                        }
+                        combinators={combinators}
+                        validator={queryValidator}
+                        controlClassnames={{
+                            queryBuilder: 'queryBuilder-branches',
+                        }}
+                        controlElements={{
+                            addRuleAction: (props) => (
+                                <AddButton {...props} label="rule" />
+                            ),
+                            addGroupAction: (props) => (
+                                <AddButton {...props} label="group" />
+                            ),
+                            combinatorSelector: CombinatorSelector,
+                            removeRuleAction: RemoveButton,
+                            removeGroupAction: RemoveButton,
+                            valueEditor: ValueEditor,
+                        }}
+                    />
+                </QueryBuilderMaterial>
+            </Grid>
+            <Grid item xs={12}>
+                <ErrorInput name={props.name} InputField={FieldErrorAlert} />
+            </Grid>
```

Comment:
Can make the error input disappeared after correcting the incorrect field in query?

Reply:
It shown systematically error when adding a new rule => to improve after merge